mpi-proxy-split/p2p_drain_send_recv.cpp

```diff
@@ -115,48 +116,46 @@ recvMsgIntoInternalBuffer(MPI_Status status, MPI_Comm comm)
 }
 
 // Go through each MPI_Irecv in the g_async_calls map and try to complete
-// the MPI_Irecv before checkpointing.
+// the MPI_Irecv and MPI_Isend before checkpointing.
 int
-completePendingIrecvs()
+completePendingP2pRequests()
 {
   int bytesReceived = 0;
   dmtcp::map<MPI_Request, mpi_async_call_t*>::iterator it;
   for (it = g_async_calls.begin(); it != g_async_calls.end();) {
     MPI_Request request = it->first;
     mpi_async_call_t *call = it->second;
-    if (call->type == IRECV_REQUEST) {
-      int flag = 0;
-      MPI_Status status;
-      MPI_Test_internal(&request, &flag, &status, false);
-      if (flag) {
-        UPDATE_REQUEST_MAP(request, MPI_REQUEST_NULL);
+    int flag = 0;
+    MPI_Status status;
+    MPI_Test_internal(&request, &flag, &status, false);
```

Comment:
Can we add a comment above this similar to your comment in this PR?  This will help others to understand the code better.  You can either insert your entire PR comment, which is very well written, or if you're searching for something shorter, perhaps:
```
// This is needed if an MPI_Isend was called earlier.  Without this,
// large messages within the same node will fail under MPICH and some
// other MPIs.  A previous call to MPI_Irecv caused only the metadata to be
// exchanged.  So, MPI_Iprobe succeeds and MPI_Irecv will later fail, unless
// we force the sending of data via MPI_Test. 
```